### ISSUE TITLE
Add homebrew install instructions?

### ISSUE THREAD

**Issue by GauntletWizard** [ksonnet/kubecfg#68]
I've published my brew formulae for kubecfg and ksonnetlib; https://github.com/GauntletWizard/homebrew-kubecfg 

Install instructions would be:
    brew tap GauntletWizard/kubecfg  #or ksonnet/kubecfg
    brew install kubecfg ksonnetlib
Add the following to your `.bash_profile`:
    export KUBECFG_JPATH="/usr/local/opt/ksonnetlib/share/ksonnet-lib"

Feel free to fork the repo and update it. Updating for a new release should be as simple as updating the tag and revision sha in the appropriate formulae.

**Comment by anguslees** [comment]
I'm not a Mac user.  Is this preferable to just making the kubecfg darwin executable available from [github releases](https://github.com/ksonnet/kubecfg/releases)?  Should I drop the executable in favour of the brew formula?  Keep both?

**Comment by GauntletWizard** [comment]
Having an official binary is still useful; Not everyone uses brew. However, it is a very popular package manager, so having it is a nice-to-have (and made showing my devs how to use it a lot easier)

**Comment by prydonius** [comment]
+1 for darwin executables

**Comment by anguslees** [comment]
> EDIT: well it looks like there are darwin builds for older releases but not 0.5.0, any ideas what happened there?

@prydonius travis happened.  I've spent _days_ debugging jsonnet_cgo and trying various toolchain combinations, and I still have no idea why "go test" frequently hangs on travis' osx environment in the first(?) test case that invokes jsonnet.  I'm assured by mac users that this does not happen for them locally when they run the same commands.  It also suddenly started happening on travis without a change in the kubecfg code/setup/vendored-libs - so I'm inclined to blame a travis env issue, but I have also failed to fix it by forcing the build environment to various old versions of tools or base image.  I have also stared at the jsonnet_cgo source, and fixed several memory leaks and one potential go-garbage-collection race, all which also failed to unblock the kubecfg osx tests.

So far I manually re-trigger the osx release builds with my fingers crossed and I very occasionally get a successful build.  Better solutions are:
- cross-compile from linux->osx.  Downside of this is that we can't _test_ the osx binary - so if there actually was a real deadlock issue then we wouldn't know.
- switch to jsonnet-go, on the (untested) assumption that this is a cgo toolchain issue.

**Comment by anguslees** [comment]
To the broader homebrew question, I still need advice from a Mac user as to what I'm expected to do here.  

I _think_ I need to upload the homebrew recipe to the main homebrew repo, and then continue to manually update the embedded version number with every kubecfg release - is that correct?

**Comment by prydonius** [comment]
> So far I manually re-trigger the osx release builds with my fingers crossed and I very occasionally get a successful build. Better solutions are:

This perhaps sounds like a different issue, so don't really want to hijack, but now that jsonnet-go is feature complete, it might make sense to move to that. cgo certainly has some limitations (e.g. with static linking) that it would be nice to avoid.

**Comment by aecolley** [comment]
Mac user here. Homebrew is popular but not universal, so retaining the binaries is a reasonable thing to do. The [usual maintenance procedure with Homebrew](https://docs.brew.sh/Formula-Cookbook#updating-formulae) is to update the version number and sha256, then file a pull request. There are many formulae which lag the real source repository by a day or three because they're maintained by someone other than the real source maintainer.

Also, I just filed a PR for kubecfg (Homebrew/homebrew-core#25737), since there wasn't an existing one.

**Comment by andresmgot** [comment]
I submitted a review for the brew formula. It LGTM but you will need the approval from brew maintainers.

**Comment by aecolley** [comment]
Homebrew now has `kubecfg`. The typicall install instructions are:

1. Install [Homebrew](https://brew.sh/).
2. `brew install kubecfg`
